File: C/seleccion_secuencial.c
// Implementacion de busqueda secuencial y ordenamiento por seleccion

#include <stdio.h>

// Funcion para llenar un arreglo

void fill_array(int arr[], int size)
{
    register int i;
    int element;

    for (i = 0; i < size; i++)
    {
        printf("\nIngrese el valor del elemento %d ", i);
        scanf("%d", &element);
        arr[i] = element;
    }
    printf("\n");
}

// Funcion para imprimir un arreglo

void print_array(int arr[], int size)
{
    register int i;

    printf("\nLos elementos del arreglo son: [ ");
    
    for (i = 0; i < size; i++)
    {
        printf("%d ",arr[i]);
    }
    
    printf("]\n");
}

// Funcion para buscar un elemento de forma secuencial

int linear_search(int arr[], int size, int v)
{
    register int i;

    for (i = 0; i < size; i++)
    {
        if (arr[i] == v)
            return i;
    }
    return -1;
}

// Funcion para implementar selection sort

void selection_sort(int arr[], int size)
{
    register int i, j;
    int min;

    for (i = 0; i < size - 1; i++)
    {
        min = i;
        for (j = i + 1; j < size; j++)
        {
            if (arr[j] < arr[min])
                min = j;
        }
        if (min != i)
        {
            int aux;
            aux = arr[i];
            arr[i] = arr[min];
            arr[min] = aux;
        }
    }
}

void main()
{
    int pos, n, value;

    printf("\nIngrese la cantidad de elementos del arreglo: ");
    scanf("%d", &n);

    int array[n];

    // Llamada a funcion para rellenar el arreglo

    fill_array(array, n);
    print_array(array, n);

    // Llamada a funcion para buscar un elemento

    printf("\nIngrese el valor del elemento a buscar: ");
    scanf("%d", &value);

    pos = linear_search(array, n, value);

    printf("\nLa posicion del elemento es: %d\n", pos);

    // Llamada a funcion para ordenar el arreglo por seleccion

    selection_sort(array, n);
    printf("\nOrdenando el arreglo por seleccion...\n");
    print_array(array, n);
}

File: Python/seleccion_secuencial.py
# Programa que busca un elemento de forma secuencial y ordena la lista usando selection sort

# Funcion para llenar una lista

def fill_list(lista):
    length = int(input("\nIngrese la cantidad de elementos de la lista: "))

    for i in range(0, length):
        value = int(input(f"Ingrese el elemento numero {i}: "))
        lista.append(value)

# Funcion para buscar un elemento de forma secuencial

def linear_search(lista, size, v):

    for i in range(0, size):
        if lista[i] == v:
            return i
    
    return -1

# Funcion para ordenar la lista usando selection sort

def selection_sort(lista, size):
    for i in range(0, size - 1):
        min = i
        for j in range(i + 1, size):
            if lista[j] < lista[min]:
                min = j

        if min != i:
            aux = lista[i]
            lista[i] = lista[min]
            lista[min] = aux



# Programa main()

int_list = []

# Llamada a funcion para llenar la lista

fill_list(int_list)
list_size = len(int_list)
print(f"\nLa lista es: {int_list}")

# Llamada a funcion para buscar un elemento en la lista

value = int(input("\nIngrese el elemento a buscar: "))
pos = linear_search(int_list, list_size, value)
print(f"\nLa posicion del elemento {value} es: {pos}")

# Llamada a funcion para ordenar la lista usando selection sort

print("\nOrdenando mediante selection sort...")
selection_sort(int_list, list_size)
print(f"\nLa lista ordenada es: {int_list}\n")